### DISCUSSION TITLE
Is there a way to ignore specific links for robot-world collisions?

### DISCUSSION THREAD

**Discussion by scharalambous-sony** [NVlabs/curobo#191] [Software Q&A] [answered]
Hi,
The base and another link of my robot will never collide with the world (world is static)
How can I exclude specific links from the robot-world collision checks?
I still need them for self-collision checking. I'm also aware of the possibility to use smaller spheres for those links and then use self_collision_buffer, but that's not possible in my case as the center of some spheres lies inside the world collision representation

I understand there is no existing feature for this so any pointer to where I can implement it will be helpful!

**Comment by balakumar-s** [comment, marked as answer]
If the sphere radius is set to a negative value, then collision checking is ignored for that sphere. 

You can set a negative radius and then add a value to self collision buffer to enable only self collision checking.

**Comment by scharalambous-sony** [comment]
What I ended up doing was adding a list of strings named world_collision_ignore in the robot yml file and added these lines to [CudaRobotGenerator](https://github.com/NVlabs/curobo/blob/main/src/curobo/cuda_robot_model/cuda_robot_generator.py#L595):

```
            for j_idx, j in enumerate(collision_link_names): 
                if j in self.world_collision_ignore:
                    continue
```
as well as a corresponding member in the constructor
Thanks though

**Reply by balakumar-s** [reply]
This would ignore the links entirely as the spheres for the ignored links would never get created. Diss you check if it avoids self collisions?

**Reply by scharalambous-sony** [reply]
I see, the _link_spheres_tensor is used for self collision distance too. My code above is wrong then. I will use the negative radius approach
